Project/Sources/Methods/_onWebConnection.4dm

```diff
@@ -1,20 +1,20 @@
 //%attributes = {"invisible":true}
 C_TEXT($1; $2; $3; $4; $5; $6)
 
-var $redirectURI : Text
+var $redirectURI; $state : Text
 var $responseFile; $customResponseFile; $customErrorFile : 4D.File
 
 $responseFile:=Folder(fk resources folder).file("Response_Template.html")
 
-If (OB Is defined(Storage; "params"))
-	Use (Storage.params)
-		$redirectURI:=String(Storage.params.redirectURI)
-		If (Length($redirectURI)>0)
-			$redirectURI:=_getPathFromURL($redirectURI)+"@"
-		End if 
-		$customResponseFile:=(Value type(Storage.params.authenticationPage)#Is undefined) ? Storage.params.authenticationPage : Null
-		$customErrorFile:=(Value type(Storage.params.authenticationErrorPage)#Is undefined) ? Storage.params.authenticationErrorPage : Null
-	End use 
+$state:=_getURLParameterValue($1; "state")
```

Comment:
all seems to be a pretty good refactor of the code with new 4d code features
but here is real feature change

just as an advice prefer to use separate pull request (to know if bug comme refactor, or new features things)